### PR TITLE
Fixes #24977: Create a system-updates module in Rust

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/24977

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/rudderc/Cargo.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ license.workspace = true
 [dependencies]
 anyhow = "1"
 askama = "0.12"
-boon = "0.5.0"
+boon = "0.6.0"
 clap = { version = "4", features = ["derive"] }
-quick-xml = { version = "0.31.0", features = ["serialize"] }
+quick-xml = { version = "0.36.0", features = ["serialize"] }
```
Comment (Fdall) [inline policies/rudderc/Cargo.toml:16]: This seems tol break the metadata generation

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/rudderc/Cargo.toml (hunk 1 above)
Comment (amousset) [reply policies/rudderc/Cargo.toml:16]: indeed (it is 0.32 with the changes in escaping)

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request closed without merging**